compiler/src/dotty/tools/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -110,7 +166,12 @@ class BTypesFromSymbols[I <: BackendInterface](val int: I) extends BTypes {
       // For consistency, the InnerClass entry for D needs to be present in C - to Java it looks
       // like D is a member of C, not C$.
       val linkedClass = classSym.linkedClass
-      val companionModuleMembers = classSym.companionModuleMembers
+      val companionModuleMembers = {
+        // phase travel to exitingPickler: this makes sure that memberClassesOf only sees member classes,
+        // not local classes of the companion module (E in the example) that were lifted by lambdalift.
+        if (classSym.linkedClass.isTopLevelModuleClass) /*exitingPickler*/ getMemberClasses(classSym.linkedClass)
```

Comment:
The comment and the commented out call to `exitingPickler` can be removed, getMemberClasses takes care of using the correct phase.

Reply:
Done